src/commands/coin/coinUpdate.ts

```diff
@@ -22,17 +30,32 @@ class CoinBalanceCommand extends AdminCommand {
           key: 'amount',
           prompt: 'enter the new balance.',
           type: 'integer'
+        },
+        {
+          key: 'reason',
+          prompt: '',
+          type: 'string',
+          default: ''
         }
       ]
     });
   }
 
-  async onRun(message: CommandoMessage, args: { user: User; amount: number }): Promise<Message> {
-    const { user, amount } = args;
+  async onRun(message: CommandoMessage, args: { user: User; amount: number; reason: string }): Promise<Message> {
+    const { user, amount, reason } = args;
+    // Get old balance
+    const oldBalance = await getCoinBalanceByUserId(user.id);
     // Update coin balance
     await updateCoinBalanceByUserId(user.id, amount);
     // Get new balance
     const newBalance = await getCoinBalanceByUserId(user.id);
+    await updateUserCoinLedger(
```

Comment:
adding entry to the ledger here is not a good idea, you should add entries to the ledger in the two update/adjust backend APIs, since this backend logic. you can just pass down an optional reason param to those functions if needed.